universum/modules/launcher.py

```diff
@@ -282,14 +272,12 @@ def finalize(self) -> None:
                 text = utils.trim_and_convert_to_unicode(text)
                 if self.file:
                     self.file.write(text + "\n")
-                self.add_tag(self.configuration.fail_tag)
                 self._error = text
-                return
-
-            self.add_tag(self.configuration.pass_tag)
-            return
 
         finally:
+            tag: Optional[str] = self._get_teamcity_build_tag()
```

Comment:
For me the benefit of having two functions and storing the result in a variable is not clear

Reply:
Ok, just for reference: the returned tag can be None even without additional check of condition, just because the tag is not set in the configuration. This makes the introduction of variable more reasonable.